src/pycrunch/importing.py

```diff
@@ -52,6 +52,40 @@ def wait_for_batch_status(self, batch, status):
         else:
             raise ValueError("The batch did not reach the '%s' state in the "
                              "given time. Please check again later." % status)
+    
+    def add_schema_metadata(self, site, schema, metadata, filename, fp, mimetype="application/x-parquet"):
+        """
+        Create a new Source from a parquet file using schema and metadata.
+
+        Parameters:
+        site (shoji.Catalog): a shoji Catalog object, from which we acquire session and sources url
+        schema (str): json string containing schema
+        metadata (str): json string containing metadata
+        filename (str): name of file being uploaded
+        fp (BufferedReader): opened file object
+        mimetype (str): mimetype of file being uploaded
+
+        Returns:
+        shoji.Entity: Shoji entity containing the payload, status_code and source_url of uploaded file
+        """
+        response = site.session.post(
+            site.catalogs.sources, 
+            files={
+                "uploaded_file": (filename, fp, mimetype)
+            },
+            data={
+                "schema": schema, 
+                "metadata": metadata,
+                "crunchlake": "create",
+                "dataset_id": "None"
+            }
+        )
+        return shoji.Entity(site.session, body={
+                "status_code": response.status_code,
```

Comment:
These aren't the attributes of a new Source entity. 

These `body` attributes should match the same as if you were reading a Source from the API. At this point right after the creation you won't have them. The response will probably be a 201 with a Location header.

So, the only attribute you should be able to fill form there is the `self`.

See how it's being done here:
https://github.com/Crunch-io/pycrunch/blob/master/src/pycrunch/shoji.py#L180
```
entity["self"] = URL(seeother[-1].headers["Location"], "")
```
The `Entity` you return is ok to be empty as long as it has a `self` url.

Then the consumer will be able to do `source.refresh()` to trigger a new GET request and fetch all the attributes if they want.